pa-rserve/src/main/java/org/ow2/parserve/util/rsession/StartRserve.java

```diff
@@ -0,0 +1,253 @@
+                    out.append(output.getOutput());
+                }
+                if (err != null) {
+                    err.append(error.getOutput());
+                }
+            }
+        } catch (Exception x) {
+            logger.error(x);
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * shortcut to <code>launchRserve(cmd, "--no-save --slave", "--no-save --slave", false)</code>
+     */
+    public static boolean launchRserve(String cmd) {
+        return launchRserve(cmd, /*null,*/ "--no-save --slave", null, "--no-save --slave", false, false, -1);
+    }
+
+    /**
+     * attempt to start Rserve. Note: parameters are <b>not</b> quoted, so avoid using any quotes in arguments
+     *
+     * @param cmd      command necessary to start R
+     * @param rargs    arguments are are to be passed to R
+     * @param rsrvargs arguments to be passed to Rserve
+     * @return <code>true</code> if Rserve is running or was successfully started, <code>false</code> otherwise.
+     */
+    public static boolean launchRserve(String cmd, /*String libloc,*/ String rargs, File preloadFile, String rsrvargs, boolean daemon, boolean debug, int timeout) {
+        logger.info("Waiting for Rserve to start ...");
+        boolean startRserve;
+        if (daemon) {
+            startRserve = doInR("library(Rserve);" + (preloadFile != null ? "source('" + Utils.toRpath(preloadFile) + "');" : "") + "Rserve(" + (debug ? "TRUE" : "FALSE") + ",args='" + rsrvargs + "')", cmd, rargs, false, null, null);
+        } else {
+            startRserve = doInR("library(Rserve);" + (preloadFile != null ? "source('" + Utils.toRpath(preloadFile) + "');" : "") + "run.Rserve(" + rsrvargs + ")", cmd, rargs, false, null, null);
+        }
+        if (startRserve) {
+            logger.info("Rserve startup done, let us try to connect ...");
+        } else {
+            return false;
+        }
+
+        // try up to 120 seconds before giving up. We can be conservative here, because at this point the process execution
+        // itself was successful and the startup is usually asynchronous
+        int attempts = (timeout > 0 ? timeout : 120);
+        while (attempts > 0) {
+            try {
+                RConnection c = null;
+                int port = -1;
+                if (rsrvargs.contains("--RS-port")) {
+                    String rsport = rsrvargs.split("--RS-port")[1].trim().split(" ")[0];
+                    port = Integer.parseInt(rsport);
+                } else if (rsrvargs.contains("port")) {
+                    String rsport = rsrvargs.split("port")[1].trim().split("=")[1].trim().split(",")[0].trim();
+                    port = Integer.parseInt(rsport);
+                }
+                logger.info("Trying to connect to RServe on port : " + port);
+                if (rsrvargs.contains("--RS-port") || rsrvargs.contains("port")) {
+                    c = new RConnection("localhost", port);
```

Comment:
the connection should be localhost only. The port parsing is error-prone and I should probably find another way.